src/components/legend/draw.js

```diff
@@ -620,11 +658,12 @@ function computeLegendDimensions(gd, groups, traces) {
 
                 groupHeight += textHeight;
             });
-
-            opts._height = Math.max(opts._height, groupHeight);
         });
 
-        opts._height += 10 + borderwidth * 2;
+        opts._height = 10 + (borderwidth * 2) + (rowNum * maxHeight) + ((rowNum - 1) * traceGap);
```

Comment:
Thanks very much for the PR @nathanjenx ! After playing around on your branch, this here doesn't appear quite right. Is `legend.tracegroupgap` taken into account?

From your new `legendgroup_horizontal_wrapping` mock, I expect that calling `Plotly.relayout(gd, 'legend.tracegroupgap', 40)` would increase the bottom margin. But now, the legend is clipped:

![image](https://user-images.githubusercontent.com/6675409/54546729-aaafda80-497a-11e9-92d6-8c44d009a82b.png)

Reply:
Thank you so much @etpinard for catching this one! It should definitely be `opts.tracegroupgap` instead of `traceGap` at the end of line 663.

I'll let @nathanjenx fix!

Reply:
@antoinerg quite right, that was the case.
@etpinard, @antoinerg I've refactored the code to make it more immediately obvious where the values come from!

Reply:
Thanks @nathanjenx for the quick fix! :tada: 

Would you be interested in adding a jasmine test making sure `relayout(gd, 'legend.tracegroupgap', ??)` makes the bottom margin resize accordingly?

If so, add an `it` block to this [`describe`](https://github.com/plotly/plotly.js/blob/fc236e2802cdfe60fee1320de55cf80af2b11eda/test/jasmine/tests/legend_test.js#L593-L708) block, asserting `gd._fullLayout._size` should be enough. You can also try to assert the legend bounding box to :lock: this down. Thanks very much for your hard work :muscle:

Reply:
No news from @nathanjenx 

@antoinerg could you take over this one to bring it to the finish line? :checkered_flag:

You should be able to push to this PR after:

```
git remote add nathanjenx git@github.com:nathanjenx/plotly.js.git
git fetch nathanjenx fix1913-horizontal-legened-groups:fix1913-horizontal-legened-groups
git checkout fix1913-horizontal-legened-groups

# then
git push nathanjenx fix1913-horizontal-legened-groups
```

Reply:
Done in 131145d

Thanks again @nathanjenx for the great work :tada: